Framework/Algorithms/src/XDataConverter.cpp

```diff
@@ -103,7 +103,7 @@ void XDataConverter::setXData(API::MatrixWorkspace_sptr outputWS,
         }
       }
     }
-    outputWS->setX(index, m_cachedX);
+    outputWS->setSharedX(index, m_cachedX);
   } else {
     const MantidVec &xBoundaries = inputWS->readX(index);
     MantidVec &xPoints = outputWS->dataX(index);
```

Comment:
Also: The function body of `ConvertToHistogram::calculateXPoints` can be replaced by a call to `VectorHelper::convertToBinBoundary`.